src/decode-arp.h

```diff
@@ -0,0 +1,40 @@
+/* Copyright (C) 2024 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ *
+ * \author Giuseppe Longo <giuseppe@glongo.it>
+ */
+
+#ifndef __DECODE_ARP_H__
+#define __DECODE_ARP_H__
+
+#define ARP_HEADER_LEN 28
+
+typedef struct ARPHdr_ {
+    uint16_t hw_type;
+    uint16_t proto_type;
+    uint8_t hw_size;
+    uint8_t proto_size;
+    uint16_t opcode;
+    uint8_t source_mac[6];
+    uint8_t source_ip[4];
+    uint8_t dest_mac[6];
+    uint8_t dest_ip[6];
```

Comment:
Why is this IP field 6 bytes? It should be 4, I think.

Additionally, reading the RFC it seems that `proto_size` determines the size here. There are a lot of ARP types. Perhaps the decoder needs to gracefully reject handling unsupported types.

Reply:
Since it essentially only supports hw_type ethernet and proto_type IPV4, lets check for that early in the decoder and set a "unsupported type" event and increment a counter for other types.

Reply:
We should clearly distinguish between the ARP types that are decoded and those that are not.
What do you think about naming the counter for the latter `decoder.unsupported_arp`?

Reply:
Wonder if we need a counter, as we can set an event and have the event as counter. It seems this is how other decoders do it.
```
❯ git grep -E "decoder.*unsupported"
src/decode-events.c:            "decoder.ltnull.unsupported_type",
src/decode-events.c:            "decoder.erspan.unsupported_version",
src/decode-events.c:            "decoder.nsh.unsupported_version",
src/decode-events.c:            "decoder.nsh.unsupported_type",
```

Reply:
Let's set an event then.